ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderElection.java

```diff
@@ -333,7 +333,7 @@ private boolean askForVotes(Phase phase, int round) throws InterruptedException,
     LOG.info("{} {} round {}: submit vote requests at term {} for {}", this, phase, round, electionTerm, conf);
     final ResultAndTerm r = submitRequestAndWaitResult(phase, conf, electionTerm);
     LOG.info("{} {} round {}: result {}", this, phase, round, r);
-
+    CompletableFuture<Void> future = null;
```

Comment:
It is better to use `final CompletableFuture<Void> future;`. Then `future.join()` will never have NPE.

Reply:
Thank you for your valuable advice